release/RELEASE_PROCESS.rst

```diff
@@ -1,45 +1,127 @@
 Release Process
 ===============
 
-This document describes the process for creating new releases.
-
+The following documents the Ray release process. Please use the
+`Release Checklist`_ to keep track of your progress, as it is meant
+to be used alongside this process document. Also, please keep the
+team up-to-date on any major regressions or changes to the timeline
+via emails to the engineering@anyscale.com Google Group.
+
+Before Branch Cut
+-----------------
+1. **Create a document to track release-blocking commits.** These may be pull
+   requests that are not ready at the time of branch cut, or they may be
+   fixes for issues that you encounter during release testing later.
+   The only PRs that should be considered release-blocking are those which
+   fix a MAJOR REGRESSION (P0) or deliver an absolutely critical piece of
+   functionality that has been promised for the release (though this should
+   be avoided where possible).
+   You may make a copy of the following `template <https://docs.google.com/spreadsheets/d/1qeOYErAn3BzGgtEilBePjN6tavdbabCEEqglDsjrq1g/edit#gid=0>`_.
```

Comment:
Make sure that this is a publicly-accessible document with no internal comments on it